### PR TITLE
feat: admin export food rescues and checkins

### PR DESCRIPTION
**Pull request by Shehryar21**
## Implementation description. How did you make this change?
* Added `generateCSV` utils file back

## Steps to test
1. Create some checkins and schedules
2. Go to http://localhost:3000/admin/check-ins and click on export. It should download a csv file. Verify that the data outputted in the csv file is accurate.
3. Go to http://localhost:3000/admin/view-donations and click on export. It should download a csv file. Verify that the data outputted in the csv file is accurate.

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages follow conventional commits and are descriptive. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s) 
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR
- [ ] The appropriate tests if necessary have been written

### CODE REVIEW

File: frontend/src/components/pages/AdminDashboard/ViewDonationsAndCheckIns.tsx [hunk 1]
```diff
@@ -127,6 +131,11 @@ const ViewDonationsAndCheckIns = ({
     ]);
   };
 
+  const handleCSVDownload = async () => {
+    const csvScheduleData = await getScheduleCSVData();
+    downloadCSV(csvScheduleData, "foodRescues");
+  };
```
Comment (hanlinc27) [inline frontend/src/components/pages/AdminDashboard/ViewDonationsAndCheckIns.tsx:137]: nit: why don't we keep this function inside of `CSVUtils` and expose a parameter to download the correct checkin/scheduling csv?

File: frontend/src/components/pages/AdminDashboard/getCSVData.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import { generateCSV } from "../../../utils/CSVUtils";
+
+export const getScheduleCSVData = async () => {
+  const schedules = await SchedulingAPIClient.getSchedules();
+  const csvData = await Promise.all(
+    schedules.map(async (schedule, id) => {
+      let volunteer;
+      if (schedule.volunteerId) {
+        volunteer = await VolunteerAPIClient.getVolunteerById(
+          schedule.volunteerId!.toString(),
```
Comment (hanlinc27) [inline frontend/src/components/pages/AdminDashboard/getCSVData.tsx:14]:
```suggestion
          schedule.volunteerId.toString(),
```

**Review by hanlinc27 — CHANGES_REQUESTED** [review]
Functionality works for me. From the admin's perspective, these dates are pretty hard to read - can we format this properly when it's being mapped in `getCSVData`?
Screen Shot 2022-04-14 at 4 36 33 PM

also two other questions
1. Should we be fetching _all_ checkins/schedules or just upcoming?
2. Is it worth it to semi optimize the fetching function to store volunteerId and donorId's in a hashmap or something so that we're not fetching it from the backend for every single entry?

**Comment by Shehryar21** [comment]
> Functionality works for me. From the admin's perspective, these dates are pretty hard to read - can we format this properly when it's being mapped in `getCSVData`? Screen Shot 2022-04-14 at 4 36 33 PM
> 
> also two other questions
> 
> 1. Should we be fetching _all_ checkins/schedules or just upcoming?
> 2. Is it worth it to semi optimize the fetching function to store volunteerId and donorId's in a hashmap or something so that we're not fetching it from the backend for every single entry?

1) We should be fetching all (confirmed with Amy)
2) tbh dont think its necessary. Given that fetching only happens on export, it wont happen that often anyways.

**Review by hanlinc27 — APPROVED** [review]

**Pull request merged**